packages/grid/src/GridUtils.ts

```diff
@@ -126,6 +102,20 @@ export class GridUtils {
     };
   }
 
+  static getColumnHeaderDepthAtY(
+    y: Coordinate,
+    metrics: GridMetrics
+  ): number | undefined {
+    const row = GridUtils.getRowAtY(y, metrics);
+    const { columnHeaderHeight, columnHeaderMaxDepth } = metrics;
+
+    if (row === null && y <= columnHeaderHeight * columnHeaderMaxDepth) {
+      return columnHeaderMaxDepth - Math.ceil(y / columnHeaderHeight);
```

Comment:
Need to max this at `columnHeaderMaxDepth - 1`, right now if you come in from the very top you can get a depth index larger than the max index.